Tasks/Gradle/CodeAnalysis/Common/CodeAnalysisResultPublisher.ts

```diff
@@ -101,28 +101,31 @@ export class CodeAnalysisResultPublisher {
     private createSummaryContent(): string {
 
         var buildSummaryLines: string[] = [];
-        var resultsGroupedByTool: AnalysisResult[][] = this.groupBy(this.analysisResults, (o: AnalysisResult) => { return o.toolName; });
+        var resultsGroupedByTool: AnalysisResult[][] =
+            this.groupBy(this.analysisResults, (o: AnalysisResult) => { return o.originatingTool.toolName; });
 
         for (var resultGroup of resultsGroupedByTool) {
             var summaryLine = this.createSummaryLine(resultGroup);
-            buildSummaryLines.push(summaryLine);
+            if (summaryLine != null) {
```

Comment:
Defensive coding - when the build summary lines are joined nulls should be ignored anyway, but better to err on the side of caution.